### PR TITLE
System package no hardcode

### PR DESCRIPTION
**Pull request by ashcrow**
Fixes for https://bugzilla.redhat.com/show_bug.cgi?id=1450307 to avoid atomic/rpmbuild issue.

### CODE REVIEW

**Comment by giuseppe** [comment]
LGTM

@ganhuang does it solve the problem you've observed?

**Comment by ashcrow** [comment]
aos-ci-test

**Comment by ganhuang** [comment]
@giuseppe , amm, Looking the code, I can't see what changes are made for the `atomic pull ...` command which is called in the end by the ansible. I suspect this should be a `atomic` issue that is replying on the `rpm-build` package. Anyway, I haven't tested it.

**Comment by ashcrow** [comment]
@ganhuang The main change here is that ``'--system-package=no`` is added to the atomic install command inside the module. Previously we were using ``system-package=no`` in the ``values`` parameter. This used the ``set`` directive but that, mixed with the latest atomic command, caused a problem.

File: roles/lib_openshift/library/oc_atomic_container.py [hunk 1]
```diff
@@ -73,7 +73,9 @@
 def _install(module, container, image, values_list):
     ''' install a container using atomic CLI.  values_list is the list of --set arguments.
     container is the name given to the container.  image is the image to use for the installation. '''
-    args = ['atomic', 'install', "--system", '--name=%s' % container] + values_list + [image]
+    # NOTE: system-package=no is hardcoded. This should be changed to an option in the future.
+    args = ['atomic', 'install', '--system', '--system-package=no',
```
Comment (ashcrow) [inline roles/lib_openshift/library/oc_atomic_container.py]: @ganhuang this line is where we are hardcoding in the ``--system-package=no``.

File: roles/lib_openshift/src/ansible/oc_atomic_container.py [hunk 2]
```diff
@@ -9,7 +9,9 @@
 def _install(module, container, image, values_list):
     ''' install a container using atomic CLI.  values_list is the list of --set arguments.
     container is the name given to the container.  image is the image to use for the installation. '''
-    args = ['atomic', 'install', "--system", '--name=%s' % container] + values_list + [image]
+    # NOTE: system-package=no is hardcoded. This should be changed to an option in the future.
+    args = ['atomic', 'install', '--system', '--system-package=no',
```
Comment (ashcrow) [inline roles/lib_openshift/src/ansible/oc_atomic_container.py]: @ganhuang this another location where we are hardcoding in the ``--system-package=no``.

**Comment by ganhuang** [comment]
@ashcrow Thanks for the explanation. This change LGTM. But I don't think it's fixing `BZ1450307`. As you can see `Comment 11` in that bug, I tried to to run the command with "system-package=no" on the failed host manually, it still failed. I don't think there's any workarounds here until https://github.com/projectatomic/atomic/pull/1005 is fixed.

**Comment by ashcrow** [comment]
@ganhuang I could be wrong but the examples in the bug look like they were generating "system-package=no" instead of "--system-package=no".

**Comment by ashcrow** [comment]
@ganhuang as an example:

```
atomic install --storage=ostree --system container-engine:latest --name=container-engine "system-package=no"
```

versus

```
atomic install --storage=ostree --system container-engine:latest --name=container-engine --system-package=no"
```

**Comment by ganhuang** [comment]
@ashcrow  It's still failed: (have no ideas to debug ansible module)
```
fatal: [openshift-xxx.lab.sjc.redhat.com]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "invocation": {
        "module_args": {
            "image": "brew-xxxxxxxx.redhat.com:8888/rhel7/container-engine:latest", 
            "name": "container-engine", 
            "state": "latest", 
            "values": []
        }
    }
}

MSG:

No JSON object could be decoded

```

**Comment by ashcrow** [comment]
@ganhuang thanks for testing. I didn't get this error when I ran through it yesterday but I'll try to reproduce it today.

**Comment by ashcrow** [comment]
So far I've not been able to replicate the error but I'll keep trying.

@ganhuang can you verify what version of ``atomic`` is being pulled down?

**Comment by ashcrow** [comment]
@ganhuang @giuseppe It took a while but I found the issue. It looks like if no other containers have been installed the following in the ``oc_atomic_container`` module fails:

```python
args = ['atomic', 'containers', 'list', '--json', '--all', '-f', 'container=%s' % name]
    rc, out, err = module.run_command(args, check_rc=False)
    if rc != 0:
        module.fail_json(rc=rc, msg=err)
        return

containers = json.loads(out)
```

The reason is the ``atomic`` command is not giving a valid json response when no containers are present.

I've added a commit to workaround this and will open a bug with ``atomic`` so it can be fixed at a later date.

**Comment by ashcrow** [comment]
aos-ci-test

**Comment by ashcrow** [comment]
How I tested the module changes

```yaml
- hosts: vm_test_host
  gather_facts: no
  roles:
    - lib_openshift
  tasks:
    # NOTE: no_proxy added as a workaround until https://github.com/projectatomic/atomic/pull/999 is released
    - name: Pre-pull Container Engine System Container image
      command: "atomic pull --storage ostree brew-xxxxxxxx.redhat.com:8888/rhel7/container-engine:latest"
      changed_when: false
      environment:
        NO_PROXY: ""

    - name: Install Container Engine System Container
      oc_atomic_container:
        name: container-engine
        image: brew-xxxxxxxx.redhat.com:8888/rhel7/container-engine:latest
        state: latest
```

**Comment by ashcrow** [comment]
aos-ci-test

**Comment by ganhuang** [comment]
Task `Install Container Engine System Container` passed! However installer is hanging on task `Start the Container Engine service`
```
TASK [docker : Start the Container Engine service] *****************************
Friday 26 May 2017  00:42:40 +0000 (0:00:02.786)       0:03:26.592 ************ 
```
Anyway, it should not be related with this issue ʘ‿ʘ

**Comment by giuseppe** [comment]
[merge]

**Comment by giuseppe** [comment]
[merge]

**Comment by giuseppe** [comment]
aos-ci-test

**Comment by giuseppe** [comment]
aos-ci-test

**Comment by giuseppe** [comment]
[merge]

**Comment by ashcrow** [comment]
[merge][severity: blocker]

**Comment by ashcrow** [comment]
```
chroot: cannot change root directory to /rootfs: No such file or directory
```

Unrelated.

**Comment by ashcrow** [comment]
[merge][severity: blocker]

**Comment by ashcrow** [comment]
/cc @sdodson 

I could not find a flake for this.

**Comment by sdodson** [comment]
> chroot: cannot change root directory to /rootfs: No such file or directory

where'd you see that? that's a bug that needs to be fixed in origin but I don't see that in the logs here.

**Comment by sdodson** [comment]
[merge][severity: blocker]

**Comment by sdodson** [comment]
flake on https://github.com/openshift/origin/issues/13980 in https://ci.openshift.redhat.com/jenkins/job/merge_pull_request_openshift_ansible/481/

**Comment by sdodson** [comment]
we'll manually merge tomorrow morning if this fails again

**Pull request merged**